lib/sandbox/execution.js

```diff
@@ -29,14 +31,14 @@ Execution = function (id, event, context, options) {
     this.data = _.get(context, PROPERTY.DATA, {});
     this.cookies = new sdk.PropertyList(sdk.Cookie, null, context.cookies);
 
-    this.environment = sdk.VariableScope.isVariableScope(context.environment) ?
-        context.environment : new sdk.VariableScope(context.environment);
-    this._variables = sdk.VariableScope.isVariableScope(context._variables) ?
-        context._variables : new sdk.VariableScope(context._variables);
-    this.globals = sdk.VariableScope.isVariableScope(context.globals) ?
-        context.globals : new sdk.VariableScope(context.globals);
-    this.collectionVariables = sdk.VariableScope.isVariableScope(context.collectionVariables) ?
-        context.collectionVariables : new sdk.VariableScope(context.collectionVariables);
+    _.forEach(CONTEXT_VARIABLE_SCOPES, function (variableScope) {
+        // normalize variable scope instances
+        this[variableScope] = sdk.VariableScope.isVariableScope(context[variableScope]) ?
```

Comment:
We need to keep the check because all instances get serialized when sent over bridge through `uvm`. So the context variables are always JSON serialized variables instead of instances.